msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/ResetPasswordStates.kt

```diff
@@ -65,10 +66,15 @@ import java.io.Serializable
  */
 class ResetPasswordCodeRequiredState internal constructor(
     override val flowToken: String,
-    private val config: NativeAuthPublicClientApplicationConfiguration
-) : BaseState(flowToken), State, Serializable {
+    private val config: NativeAuthPublicClientApplicationConfiguration = NativeAuthPublicClientApplication.applicationConfig
+) : BaseState(flowToken), State, Parcelable {
     private val TAG: String = ResetPasswordCodeRequiredState::class.java.simpleName
 
+    constructor(parcel: Parcel) : this(
+        parcel.readString()!!
```

Comment:
Can we make this (and in other instances) a bit more elegant? The application shouldn't crash if the String returned from `readString()` is null.

Reply:
Done